pkgs/development/tools/phpactor/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib, stdenvNoCC, fetchFromGitHub, php }:
+
+let
+  version = "2023.01.21";
+
+  src = fetchFromGitHub {
+    owner = "phpactor";
+    repo = "phpactor";
+    rev = version;
+    hash = "sha256-jWZgBEaffjQ5wCStSEe+eIi7BJt6XAQFEjmq5wvW5V8=";
+  };
+
+  vendor = stdenvNoCC.mkDerivation rec {
+    pname = "phpactor-vendor";
+    inherit src version;
+
+
+    # See https://github.com/NixOS/nix/issues/6660
+    dontPatchShebangs = true;
+
+    nativeBuildInputs = [
+      php
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      substituteInPlace composer.json \
+        --replace '"config": {' '"config": { "autoloader-suffix": "Phpactor",' \
+        --replace '"name": "phpactor/phpactor",' '"name": "phpactor/phpactor", "version": "${version}",'
+      ${php.packages.composer}/bin/composer install --no-interaction --optimize-autoloader --no-dev --no-scripts
```

Comment:
Shouldn't this be in nativeBuildInputs or would that break splicing?

Reply:
it should be `phpPackages.composer` because https://github.com/NixOS/nixpkgs/issues/211340
and it should be in `nativeBuildInputs`

Reply:
What's the difference between `phpPackages.composer` and `php.packages.composer` ?

Reply:
> What's the difference between `phpPackages.composer` and `php.packages.composer` ?

nothing other than https://github.com/NixOS/nixpkgs/issues/211340

Reply:
I don't know about splicing. Is what I just pushed okay?

Reply:
yes

Reply:
though you'll have to update the outputHash on *every* change and update